pkg/controller/apigatewayv2/domainname/hooks.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package domainname
+
+import (
+	"context"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/apigatewayv2"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/apigatewayv2/v1alpha1"
+	aws "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// SetupDomainName adds a controller that reconciles DomainName.
+func SetupDomainName(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(svcapitypes.DomainNameGroupKind)
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&svcapitypes.DomainName{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.DomainNameGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithConnectionPublishers(),
```

Comment:
Do we need to override this? I am fairly certain it is just a no-op if no secret is defined

Reply:
With most of the resources not publishing anything, I thought this would be a safe default. In fact, we may want to make this default in runtime, too.

Reply:
The `APISecretPublisher` effectively accomplishes the same thing though right? So it seems like a more powerful default: "If you provide a secret we will publish it, if not we will do nothing and not error."